lxl-web/src/routes/(app)/[[lang=lang]]/(find)/FacetGroup.svelte

```diff
@@ -4,37 +4,48 @@
 	import { type FacetGroup } from './search';
 	import BiChevronRight from '~icons/bi/chevron-right';
 	import BiChevronDown from '~icons/bi/chevron-down';
+	import BiDiamond from '~icons/bi/diamond';
 
 	export let group: FacetGroup;
 	export let locale: LocaleCode;
+	export let searchPhrase = '';
 
 	const defaultFacetsShown = 5;
 	let facetsShown = defaultFacetsShown;
-	let expanded = false;
-	let searchPhrase = '';
+	let manuallyExpanded = false;
 
 	$: numfacets = group.facets.length;
+	$: hasHits = filteredFacets.length > 0;
+	$: expanded = (searchPhrase && hasHits) || (!searchPhrase && manuallyExpanded);
 	$: filteredFacets = group.facets.filter((facet) =>
-		facet.str.toLowerCase().startsWith(searchPhrase.toLowerCase())
+		facet.str
+			.toLowerCase()
+			.split(/\s|--/)
+			.find((s) => s.startsWith(searchPhrase.toLowerCase()))
 	);
 	$: shownFacets = filteredFacets.filter((facet, index) => index < facetsShown);
 	$: canShowMoreFacets = filteredFacets.length > facetsShown;
 	$: canShowLessFacets = !canShowMoreFacets && filteredFacets.length > defaultFacetsShown;
 </script>
 
-<li class="my-4 border-b-[2px] border-primary pb-2">
+<li class="my-4 border-b-[2px] border-primary pb-2" class:hidden={searchPhrase && !hasHits}>
 	<button
 		id={'toggle-' + group.dimension}
 		type="button"
-		on:click={() => (expanded = !expanded)}
+		on:click={() => {
+			manuallyExpanded = !manuallyExpanded;
+		}}
 		aria-expanded={!!expanded}
 		aria-controls={'group-' + group.dimension}
 		class="w-full text-left font-bold"
 		data-testid="facet-toggle"
 	>
 		<span class="flex items-center gap-2">
 			<span>
-				{#if expanded}
+				{#if searchPhrase}
+					<!-- FIXME -->
+					<BiDiamond class="text-icon-default" />
+				{:else if expanded}
```

Comment:
I would prefer skipping the diamond icon for now (and instead just show the chevron down icon).

It's not the scope of this PR but I think we should soon rework the facet group list to use native [`<details>` ](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) elements instead of having buttons controlling the expanded javascript state.

Reply:
Ok, replaced.
It feels a bit broken though since it is normally clickable and you still get the hand icon but now it is not clickable.